src/Mod/Draft/draftmake/make_polararray.py

```diff
@@ -125,10 +149,67 @@ def make_polar_array(base_object,
         _err(_tr("Wrong input: must be a vector."))
         return None
 
+    _msg("axis_object: {}".format(axis_object))
+    _msg("axis_edge: {}".format(axis_edge))
+    axis = axis_object
+    if axis_object and isinstance(axis_object, str):
+        found, axis = utils.find_object(axis_object,
+                                        doc=App.activeDocument())
+        if not found:
+            _err(_tr(
+                "Wrong input. Given axis_name does not refer to an "
+                "existing DocumentObject."))
+            return None
+    if axis:
+        axis_edge_name = axis_edge
+        if axis_edge:
+            if isinstance(axis_edge, str):
+                print(axis)
+                print(axis_edge)
+                edge_object = axis.getSubObject(axis_edge)
+            elif isinstance(axis_edge, int):
+                axis_edge_name = "Edge" + str(axis_edge)
+                edge_object = axis.getSubObject(axis_edge_name)
+            else:
+                _err(_tr(
+                    "Wrong input. Given axis_edge has to be of type int or str."
+                ))
+                return None
+        else:
+            if not (hasattr(axis, "Shape") and hasattr(axis.Shape, "Edges")):
+                _err(_tr("Wrong input: axis_object cannot be used for Axis Reference,"
+                         "it lacks a Shape with Edges."))
+                return None
+            axis_edge_name = "Edge1"  # this is the default if axis_edge is missing
+            edge_object = axis.getSubObject(axis_edge_name)
+        if not edge_object:
+            _err(_tr(
+                "Wrong input. Given axis_edge does not refer to a "
+                "SubObject of axis_object or given axis_object lacks"
+                "having edges."))
+            return None
+        if not isinstance(edge_object, Part.Edge):
+            _err(_tr(
+                "Wrong input. Given axis_edge does not refer to a "
+                "SubObject of type Part.Edge"))
+            return None
+        if not isinstance(edge_object.Curve, Part.Line):
+            _err(_tr(
+                "Wrong input. Given axis_edge does not refer to a "
+                "SubObject with Curve of type Part.Line"))
+            return None
+
     use_link = bool(use_link)
     _msg("use_link: {}".format(use_link))
 
     new_obj = make_array.make_array(base_object,
                                     arg1=center, arg2=angle, arg3=number,
                                     use_link=use_link)
+
+    if axis_object:
+        axis_reference = [axis, axis_edge_name]
+        _msg("axis_reference: {}".format(axis_reference))
+        new_obj.AxisReference = axis_reference
```

Comment:
By the way, currently `AxisReference` is set to `App::PropertyLinkGlobal` in `draftobjects/array.py`. Can you try using `App::PropertyLinkSubGlobal`?

In the property editor, a `Link` does not allow us to select subelements (edges), only full objects. But the `LinkSub` varieties do allow us to select edges, which is what we want.

Otherwise we can only set the edges from the Python console, or at creation time, from the Gui Command, or from the make_polar_array function. Ideally, the user should be able to change the edge also from the property editor.

Reply:
Yes that is already the case :) If you look at line 192 in `array.py` it's there.